assets/app/scripts/directives/fromFile.js

```diff
@@ -0,0 +1,397 @@
+        TaskList.clear();
+
+        $scope.aceLoaded = function(editor) {
+          aceEditorSession = editor.getSession();
+          aceEditorSession.setOption('tabSize', 2);
+          aceEditorSession.setOption('useSoftTabs', true);
+          editor.setDragDelay = 0;
+          editor.$blockScrolling = Infinity;
+        };
+
+        var checkErrorAnnotations = function() {
+          var editorAnnotations = aceEditorSession.getAnnotations();
+          $scope.editorErrorAnnotation = _.some(editorAnnotations, { type: 'error' });
+        };
+
+
+        // Determine whats the input format (JSON/YAML) and set appropriate view mode
+        var updateEditorMode = _.debounce(function(){
+          $scope.$apply(checkErrorAnnotations);
+          try {
+            JSON.parse($scope.editorContent);
+            aceEditorSession.setMode("ace/mode/json");
+          } catch (e) {
+            try {
+              YAML.parse($scope.editorContent);
+              aceEditorSession.setMode("ace/mode/yaml");
+            } catch (e) {
+              return;
+            }
+          }
+        }, 300);
+
+        // Check if the editor isn't empty to disable the 'Add' button. Also check in what format the input is in (JSON/YAML) and change
+        // the editor accordingly.
+        $scope.aceChanged = updateEditorMode;
+
+        $scope.create = function() {
+          delete $scope.alerts['create'];
+          delete $scope.error;
+          var resource;
+
+          // Trying to auto-detect what format the input is in. Since parsing JSON throws only SyntexError
+          // exception if the string to parse is not valid JSON, it is tried first and then the YAML parser
+          // is trying to parse the string. If that fails it will print the reason. In case the real reason
+          // is JSON related the printed reason will be "Reason: Unable to parse", in case of YAML related
+          // reason the true reason will be printed, since YAML parser throws an error object with needed
+          // data.
+          try {
+            resource = JSON.parse($scope.editorContent);
+          } catch (e) {
+            try {
+              resource = YAML.parse($scope.editorContent);
+            } catch (e) {
+              $scope.error = e.message;
+              return;
+            }
+          }
+
+          // Top level resource field check. 
+          if (!validateFields(resource)) {
+            return;
+          }
+
+          $scope.resourceKind = resource.kind;
+          
+          if ($scope.resourceKind.endsWith("List")) {
+            $scope.isList = true;
+            $scope.resourceList = resource.items;
+            $scope.resourceName = '';
+          } else {
+            $scope.resourceList = [resource];
+            $scope.resourceName = resource.metadata.name;
+            if ($scope.resourceKind === "Template") {
+              $scope.templateOptions = {
+                process: true,
+                add: true
```

Comment:
Missed this on earlier review, but @jwforres and @liggitt suggested add be false by default.